org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/syntaxcoloring/SemanticHighlightingCalculator.java

```diff
@@ -9,32 +9,39 @@
 package org.eclipse.xtext.ide.tests.testlanguage.syntaxcoloring;
 
 import org.eclipse.emf.ecore.EObject;
+import org.eclipse.lsp4j.SemanticTokenModifiers;
+import org.eclipse.lsp4j.SemanticTokenTypes;
 import org.eclipse.xtext.Keyword;
+import org.eclipse.xtext.TerminalRule;
 import org.eclipse.xtext.ide.editor.syntaxcoloring.DefaultSemanticHighlightingCalculator;
 import org.eclipse.xtext.ide.editor.syntaxcoloring.HighlightingStyles;
 import org.eclipse.xtext.ide.editor.syntaxcoloring.IHighlightedPositionAcceptor;
 import org.eclipse.xtext.ide.tests.testlanguage.testLanguage.TestLanguagePackage;
+import org.eclipse.xtext.ide.tests.testlanguage.testLanguage.TypeDeclaration;
 import org.eclipse.xtext.nodemodel.ICompositeNode;
 import org.eclipse.xtext.nodemodel.INode;
 import org.eclipse.xtext.nodemodel.util.NodeModelUtils;
 import org.eclipse.xtext.util.CancelIndicator;
 
 /**
- * @author Rub�n Porras Campo - Initial Contribution and API
+ * @author Rubén Porras Campo - Initial Contribution and API
  */
 public class SemanticHighlightingCalculator extends DefaultSemanticHighlightingCalculator {
 	@Override
 	protected boolean highlightElement(EObject object, IHighlightedPositionAcceptor acceptor,
 			CancelIndicator cancelIndicator) {
-		if (object.eClass() == TestLanguagePackage.Literals.TYPE_DECLARATION) {
+		if (object instanceof TypeDeclaration) {
 			ICompositeNode node = NodeModelUtils.getNode(object);
 			for (INode childNode: node.getChildren()) {
 				if (childNode.getGrammarElement() instanceof Keyword) {
 					String nodeText = childNode.getText();
 					if (nodeText.equals("extends") || nodeText.equals("type")) {
-						acceptor.addPosition(childNode.getOffset(), childNode.getLength(), HighlightingStyles.KEYWORD_ID);
+						acceptor.addPosition(childNode.getOffset(), childNode.getLength(), SemanticTokenTypes.Keyword);
```

Comment:
The highlighting calculator is meant to be agnositic to the UI frontend, being it Eclipse or LSP. Of course an implementor might chose to use SemanticTokenTypes directly, but the underlying idea is to decouple these from each other. This change here renders the approach pointless. I'm not convinced, that it is the right solution.